### ISSUE TITLE
Should nav.archIn() and nav.archOut() be able to be called without a predicate?

### ISSUE THREAD

**Issue by seddonm1** [mcollina/levelgraph#120]
Hi,
Should Navigator methods archIn() and archOut() be able to be called without a predicate?

e.g. 
```javascript
db.nav('daniele')
      .archOut()
      .archOut()
      .values(function(err, results) {}
```

I believe it should be allowed. If so here are two test scripts which the first one works and second one doesn't.

```javascript
  it('should follow and arch going out of a vertex without using a predicate',
     function(done) {
    db.nav('daniele')
      .archOut().
      values(function(err, results) {

      expect(results).to.eql([25,'marco', 'matteo']);
      done();
    });
  });

  it('should follow multiple archs, out of successive vertices, without using predicates',
     function(done) {
    db.nav('daniele')
      .archOut()
      .archOut()
      .values(function(err, results) {

      expect(results).to.eql([32,'daniele','davide']);
      done();
    });
  });
```

**Comment by mcollina** [comment]
I think so too. Would you like sending a PR?
Il giorno ven 1 mag 2015 alle 22:49 seddonm1 <notifications@github.com> ha
scritto:

**Comment by seddonm1** [comment]
Hi mcollina.

I will have a go but I must confess once the thread enters the query planner/multiple streamers loop I get a bit confused. 

I think your Hexastore approach is very elegant and suits the leveldb architecture perfectly but I am struggling to see the benefits of added complexity with the queryplanner/sortjoinstream. I think I just haven't looked at the code enough to understand it.

Is there actually a performance gain using the query planner over a simple loop of:
levelgraph.get()
loop through streamed result(s)
if not at required node then call the levelgraph.get() again with this object

**Comment by mcollina** [comment]
Yes there is, like 2-5x. Basically SortJoin walks the database only once for each pattern, vs one for matching pattern.

Try https://github.com/mcollina/levelgraph/blob/master/benchmarks/searchStream.js
vs https://github.com/mcollina/levelgraph/blob/master/benchmarks/sortSearch.js
